contributor-docs/deprecating-components.md

```diff
@@ -0,0 +1,78 @@
+As we work on the maturity of our components, we will sometimes need to build components in a parallel track/bundle without breaking existing components. Eventually, the new components would replace the old ones in the main bundle and this document explains the process around it.
+
+## Developing a draft component
+
+Start building the new component outside of the main bundle. This includes the source code of the component being under the `src/drafts` folder and the component being exported in the draft bundle (`src/drafts/index.ts`). That way, folks who
+who would like to try the draft version of the component can export it from the draft bundle.
+
+```
+import { ActionList } from "@primer/react/drafts"
+```
+
+If it is a 1:1 replacement, it's useful to keep the component name the same for consumers. Internally, we would want to, for example, call the folder name `ActionList2` and call the docs `ActionMenu v2`.
+
+## Deprecating the component
+
+Once the component is ready to be graduated from the drafts bundle to the main one, we first plan to see in which major release the component will go into deprecation and prepare for the code changes and the communications around it.
+
+### Code side of the deprecation
```

Comment:
Do we need to add documentation for what to do in a case where we aren't replacing a deprecated component with a new one as well?

We could also add to this doc later if needed.

Reply:
Hmm that is great point! I wonder if we experienced such thing in Primer before? 🤔 What would be the alternative we can recommend here if we are not able to replace a component?

Any of the below at all?
- Temporary solutions where applicable
- Guideline on how to "replicate" the component's UI and functionality without making it a Primer component
- Recommending opening PRs for teams to fix the issues that made the component not usable

Reply:
I checked through [the deprecated components in our docs](https://primer.style/react/deprecated/ActionList), and it appears that in all cases we provided an alternative component to use even if it wasn't a one-to-one update (ex: for [deprecated `ChoiceFieldset`](https://primer.style/react/deprecated/ChoiceFieldset), we recommended to use totally different components: `CheckboxGroup` or `RadioGroup`).

Perhaps we could broaden the documentation a little bit, to cover two key cases?
1. when there is a draft/experimental component replacing a component we're deprecating - this is the case already covered by these docs
2. when there is a component being deprecated that doesn't have a one-to-one replacement, but where we recommend the use of another, different component in Primer instead

Reply:
Sounds great! I wouldn't think of recommending a different component but that makes a lot of sense! 

How about this? We could add maybe just before the `Code side of the deprecation`?

> If we are not providing a one-to-one replacement for the component we are deprecating, we will be recommending the use of another, different component in Primer instead. We will make sure to communicate this in [Primer changelog](https://github.com/github/primer/discussions/categories/primer-changelog) (GitHub staff only) and deprecated component docs as well.

Reply:
Messaging looks good! Suggesting a few tweaks to reduce the [first-person perspective](https://www.grammarly.com/blog/first-second-and-third-person/):

> If there is not a one-to-one replacement for the newly deprecated component, Primer maintainers will recommend the use of another, different component in Primer instead. This will be clearly communicate in the [Primer changelog](https://github.com/github/primer/discussions/categories/primer-changelog) (GitHub staff only) and deprecated component docs.